### PR TITLE
DCS-672 Enabling info level logging and including request trace logs

### PR DESCRIPTION
**Pull request by andrewrlee**
This links requests to the current user. There's a load of different ways to do this
but this is the pattern that we use based on the form app starter.

Also removing the hsts middleware as this comes as part of helmet.

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -36,6 +37,7 @@ app.use(setupStaticContent())
 app.use(setupWebSession())
 app.use(setupAuth({ oauthApi: apis.oauthApi, tokenVerificationApi: apis.tokenVerificationApi }))
 app.use(csrf())
+app.use(setupCurrentUserAndRequestLogging({ oauthApi: apis.oauthApi, prisonApi: apis.prisonApi }))
```
Comment (andrewrlee) [inline backend/index.js:40]: * extracting out the setting of the user in res.locals out of `routes`
* extract request logging middleware out of security middleware  
* ensuring that user gets set before logging middleware so user info can be included in request logging

File: backend/middleware/currentUser.js [hunk 2]
```diff
@@ -13,12 +13,13 @@ module.exports = ({ prisonApi, oauthApi }) => async (req, res, next) => {
...
     const caseloads = req.session.allCaseloads
-    const { name, activeCaseLoadId } = req.session.userDetails
+    const { name, username, activeCaseLoadId } = req.session.userDetails
 
     res.locals.user = {
       ...res.locals.user,
       allCaseloads: caseloads,
       displayName: name,
+      username,
       activeCaseLoad: caseloads.find(cl => cl.caseLoadId === activeCaseLoadId),
```
Comment (andrewrlee) [inline backend/middleware/currentUser.js:23]: this caseload stuff will go in a future ticket

File: backend/setupCurrentUserAndRequestLogging.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const router = express.Router()
+
+module.exports = ({ prisonApi, oauthApi }) => {
+  router.use(currentUser({ prisonApi, oauthApi }))
+  router.use(async (req, res, next) => {
+    res.locals = {
+      ...res.locals,
+      currentUrlPath: req.originalUrl,
+      prisonerSearchUrl: req.session.prisonerSearchUrl,
```
Comment (andrewrlee) [inline backend/setupCurrentUserAndRequestLogging.js:15]: need to check to see if prisonerSearchUrl can be removed

File: backend/setupWebSecurity.js [hunk 4]
```diff
@@ -1,30 +1,23 @@
...
 module.exports = () => {
-  router.use(helmet({ contentSecurityPolicy: false }))
-
-  router.use(
-    hsts({
-      maxAge: sixtyDaysInSeconds,
-      includeSubDomains: true,
-      preload: true,
-    })
-  )
```
Comment (andrewrlee) [inline backend/setupWebSecurity.js:23]: hsts not required separately as part of helmet

File: backend/index.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/index.js:40]: There are interceptors you could use to enhance the app insights logs. We have a few in psh to ignore specific types of requests.

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: backend/setupCurrentUserAndRequestLogging.js (hunk 3 above)
Comment (Nimphal) [reply backend/setupCurrentUserAndRequestLogging.js:15]: It can - it's used for the offender profile, which is not relevant for this service :)

**Pull request merged**